index.html

```diff
@@ -1787,6 +1935,296 @@ <h2>
+              </li>
+              <li>If |mime_type_parsed|.type is not listed in as a top-level
+              type in [[IANA_MEDIA_TYPES]], return failure.
+              </li>
+              <li>Add |mime_type_string| → |extensions| to |accept|.
+              </li>
+            </ol>
+          </li>
+          <li>Let |file_handler:ordered map| be |ordered map| «[ "action" →
+          |url|, "name" → |item|["name"], "accept" → |accept| ]».
+          </li>
+          <li>[=Process image resources=] passing |item|["icons"],
+          |file_handler|, |manifest URL|, and "icons".
+          </li>
+          <li>The user agent MAY replace |file_handler|["icons"] with the
+          application icon.
+          </li>
+          <li>If |file_handler|["name"] doesn't [=map/exist=], the user agent
+          MAY populate a suitable name to describe the file type.
+          </li>
+          <li>Return |file_handler|.
+          </li>
+        </ol>
+      </section>
+    </section>
+    <section data-cite="file-system-access">
+      <h2>
+        <dfn>Execute a file handler launch</dfn>
+      </h2>
+      <p>
+        A {{Window/window}} has an associated {{LaunchConsumer}} <dfn>assigned
+        launch consumer</dfn>.
+      </p>
+      <p>
+        A {{Window/window}} has an associated [=list=] of {{LaunchParams}}
+        <dfn>unconsumed launch params</dfn>.
+      </p>
+      <p>
+        When a [=file type=] is registered with a web app and one or more of
+        these registered files are launched on the platform, run the following
+        steps, given the launched {{FrozenArray}} of {{FileSystemHandle}}s
+        |files:FrozenArray&lt;FileSystemHandle&gt;| and the corresponding
+        [=file handler|file_handler=].
+      </p>
+      <ol>
+        <li>Let |url| be the |file_handler|.[=file handler/action=].
+        </li>
+        <li>Let |browsing context| be the result of creating a new [=top-level
+        browsing context=].
+        </li>
+        <li>[=Navigate=] |browsing context| to |url|.
+        </li>
+        <li>Let |params:LaunchParams| be {{LaunchParams}} with
+        {{LaunchParams.files}} set to |files|.
+        </li>
+        <li>If [=assigned launch consumer=] is set, call [=assigned launch
+        consumer=] with |params| and return
+        </li>
+        <li>[=list/Append=] |params| to [=unconsumed launch params=].
+        </li>
+      </ol>
+      <section data-dfn-for="LaunchParams">
+        <h3>
+          <dfn>LaunchParams</dfn> interface
+        </h3>
+        <pre class="idl">
+          [Exposed=Window] interface LaunchParams {
+            readonly attribute FrozenArray&lt;FileSystemHandle&gt; files;
+          };
+        </pre>
+        <section>
+          <h2>
+            <dfn>files</dfn> attribute
+          </h2>
+          <p>
+            The {{LaunchParams/files}} attribute is a {{FrozenArray}} of
+            {{FileSystemHandle}}s.
```

Comment:
worth noting these handles are read only? (but can be converted to readWrite via requestPermission)

Reply:
@mkruisselbrink I tried to state this in a way that made sense, but you'll know better how to specify this.

Reply:
Note from @mkruisselbrink - I should use language from https://wicg.github.io/file-system-access/#local-file-system-permissions

Reply:
Done.